streams/src/test/java/org/apache/kafka/streams/integration/PositionCheckpointIntegrationTest.java

```diff
@@ -0,0 +1,777 @@
+    private Properties streamsConfig;
+
+    public enum StoresToTest {
+        ROCKS_KV {
+            @Override
+            public StoreSupplier<?> supplier() {
+                return Stores.persistentKeyValueStore(STORE_NAME);
+            }
+
+            @Override
+            public boolean timestamped() {
+                return false;
+            }
+
+            @Override
+            public boolean keyValue() {
+                return true;
+            }
+        },
+        TIME_ROCKS_KV {
+            @Override
+            public StoreSupplier<?> supplier() {
+                return Stores.persistentTimestampedKeyValueStore(STORE_NAME);
+            }
+
+            @Override
+            public boolean keyValue() {
+                return true;
+            }
+        },
+        ROCKS_WINDOW {
+            @Override
+            public StoreSupplier<?> supplier() {
+                return Stores.persistentWindowStore(STORE_NAME, Duration.ofDays(1), WINDOW_SIZE,
+                                                    false
+                );
+            }
+
+            @Override
+            public boolean isWindowed() {
+                return true;
+            }
+
+            @Override
+            public boolean timestamped() {
+                return false;
+            }
+        },
+        TIME_ROCKS_WINDOW {
+            @Override
+            public StoreSupplier<?> supplier() {
+                return Stores.persistentTimestampedWindowStore(STORE_NAME, Duration.ofDays(1),
+                                                               WINDOW_SIZE, false
+                );
+            }
+
+            @Override
+            public boolean isWindowed() {
+                return true;
+            }
+        },
+        ROCKS_SESSION {
+            @Override
+            public StoreSupplier<?> supplier() {
+                return Stores.persistentSessionStore(STORE_NAME, Duration.ofDays(1));
+            }
+
+            @Override
+            public boolean isSession() {
+                return true;
+            }
+        };
+
+        public abstract StoreSupplier<?> supplier();
+
+        public boolean timestamped() {
+            return true; // most stores are timestamped
+        }
+
+        public boolean global() {
+            return false;
+        }
+
+        public boolean keyValue() {
+            return false;
+        }
+
+        public boolean isWindowed() {
+            return false;
+        }
+
+        public boolean isSession() {
+            return false;
+        }
+    }
+
+    @Parameterized.Parameters(name = "cache={0}, log={1}, supplier={2}, kind={3}")
+    public static Collection<Object[]> data() {
+        LOG.info("Generating test cases according to random seed: {}", SEED);
+        final List<Object[]> values = new ArrayList<>();
+        for (final boolean cacheEnabled : Arrays.asList(false, true)) {
+            for (final boolean logEnabled : Arrays.asList(true)) {
```

Comment:
I'm only looking at configs with a logging layer.

Reply:
Thanks. Why is that?

Reply:
We had discussed this with Patrick. I think this is because only these stores are the ones that have changelog enabled. Hence, only these stores will restore their changelog into the state store and won't start with an empty store. The rationale then was that the position should match what the state store contains: if the state store is empty, the position should be empty, whereas if the state store is restored from the changelog, the position should be restored from the checkpoint file

However, I think restoring doesn't work this way. Only

Reply:
Just changed this, tests are covering all the cases now including those with logging.

Reply:
Riiight. Sorry, I forgot :)